packages/atomic-layout/src/components/Only.tsx

```diff
@@ -1,109 +1,22 @@
-  closeBreakpoint,
-  mergeAreaRecords,
-} from '@atomic-layout/core'
-
+import { BoxProps } from '@atomic-layout/core'
 import Box from './Box'
-import { withPlaceholder } from '../utils/withPlaceholder'
-
-export type BreakpointRef = string | Breakpoint
-
-export const resolveBreakpoint = (breakpointRef: BreakpointRef): Breakpoint => {
-  return typeof breakpointRef === 'string'
-    ? Layout.breakpoints[breakpointRef]
-    : breakpointRef
-}
+import useResponsiveQuery, {
+  ResponsiveQueryParams,
+} from '../hooks/useResponsiveQuery'
 
-export interface OnlyProps extends BoxProps {
-  /**
-   * Renders children only at the specified breakpoint.
-   */
-  for?: BreakpointRef
-  /**
-   * Renders children from the specified breakpoint and up,
-   * unless enclosing `to` prop is set to form a range.
-   */
-  from?: BreakpointRef
-  /**
-   * Renders children from the specified breakpoint and down,
-   * unless the openning `from` prop is set to form a range.
-   */
-  to?: BreakpointRef
-  /**
-   * Renders children everywhere except the given breakpoint range.
-   */
-  except?: boolean
-}
-
-export const createWrapper = (
-  wrapper: React.FC,
-  children: React.ReactNode,
-  props: BoxProps,
-) => (...breakpoints: Breakpoint[]) => {
-  const Placeholder = withPlaceholder(wrapper, breakpoints)
-  return <Placeholder {...props}>{children}</Placeholder>
-}
+export type OnlyProps = BoxProps & ResponsiveQueryParams
 
 const Only: React.FC<OnlyProps> = ({
   children,
   except,
-  for: exactBreakpointName,
-  from: minBreakpointName,
-  to: maxBreakpointName,
+  for: exactBreakpoint,
+  from,
+  to,
   ...restProps
 }) => {
-  const wrapWith = createWrapper(Box, children, restProps)
-
-  // Render on explicit breakpoint
-  if (exactBreakpointName) {
-    return wrapWith(resolveBreakpoint(exactBreakpointName))
-  }
-
-  const minBreakpoint = resolveBreakpoint(minBreakpointName)
-  const maxBreakpoint = resolveBreakpoint(maxBreakpointName)
-
-  // Bell, __/--\__
-  if (minBreakpoint && maxBreakpoint && !except) {
-    const mergedAreaRecord = mergeAreaRecords(
-      {
-        behavior: 'down',
-        breakpoint: maxBreakpoint,
-      },
-      {
-        behavior: 'up',
-        breakpoint: minBreakpoint,
-      },
-      false,
-    )
-
-    return wrapWith(mergedAreaRecord.breakpoint)
-  }
-
-  // Notch, --\__/--
-  if (minBreakpoint && maxBreakpoint && except) {
-    return wrapWith(
-      closeBreakpoint(minBreakpoint),
-      openBreakpoint(maxBreakpoint),
-    )
-  }
-
-  // High-pass, __/--
-  if (minBreakpoint && !maxBreakpoint) {
-    return wrapWith(openBreakpoint(minBreakpoint))
-  }
-
-  // Low-pass, --\__
-  if (!minBreakpoint && maxBreakpoint) {
-    return wrapWith(closeBreakpoint(maxBreakpoint))
-  }
-
-  // Render always when no constrains are provided
-  return <>children</>
+  const matches = useResponsiveQuery({ for: exactBreakpoint, from, to, except })
+  return matches && <Box {...restProps}>{children}</Box>
```

Comment:
Instead of rendering in a fragment and deligating the `restProps` assignment to the `withPlaceholder` (`MediaQuery`, effectively), the `Only` component _now renders a Box_, to which it assigns all the passed box props. This should not introduce any breaking changes, as it's a matter of moving the abstracted parent component up to the application surface.